src/RoadRunnerBridge/HttpFoundationWorker.php

```diff
@@ -188,7 +179,141 @@ private function timeFloat(): float
+     * @return \Closure
+     */
+    private function createFileStreamGenerator(BinaryFileResponse $response): \Closure
+    {
+        return static function () use ($response) {
+            $ref = new \ReflectionClass($response);
+            $maxlen = $ref->getProperty("maxlen")->getValue($response);
+            $offset = $ref->getProperty("offset")->getValue($response);
+            $chunkSize = $ref->getProperty("chunkSize")->getValue($response);
+            $deleteFileAfterSend = $ref->getProperty("deleteFileAfterSend")->getValue($response);
+
+            try {
+                if (!$response->isSuccessful()) {
+                    return;
+                }
+
+                $file = fopen($response->getFile()->getPathname(), "r");
+
+                ignore_user_abort(true);
+
+                if ($maxlen === 0) {
+                    return;
+                }
+
+                if ($offset !== 0) {
+                    fseek($file, $offset);
+                }
+
+                $length = $maxlen;
+                while ($length && !feof($file)) {
+                    $read = $length > $chunkSize || 0 > $length ? $chunkSize : $length;
+
+                    if (false === $data = fread($file, $read)) {
+                        break;
+                    }
+
+                    while ('' !== $data) {
+                        try {
+                            yield $data;
+                        } catch (StreamStoppedException) {
+                            break 2;
+                        }
+
+                        if (0 < $length) {
+                            $length -= $read;
+                        }
+                        $data = substr($data, $read);
+                    }
+                }
+
+                fclose($file);
+            } finally {
+                if ($deleteFileAfterSend && is_file($response->getFile()->getPathname())) {
+                    unlink($response->getFile()->getPathname());
+                }
+            }
+        };
+    }
+
+    /**
+     * @param SymfonyResponse $response
+     * @return \Closure
+     */
+    private function createDefaultContentGetter(SymfonyResponse $response): \Closure
+    {
+        return static function () use ($response) {
+            ob_start();
+            $response->sendContent();
+            return ob_get_clean();
+        };
+    }
+
+    /**
+     * StreamedResponse callback can now use `yield` to be really streamed
+     * @param StreamedResponse $response
+     * @return \Closure
+     */
+    private function createStreamedResponseGenerator(StreamedResponse $response): \Closure
+    {
+        return function () use ($response): \Generator {
+            $kernelCallback = $response->getCallback();
+
+            $kernelCallbackRef = new \ReflectionFunction($kernelCallback);
+            $closureVars = $kernelCallbackRef->getClosureUsedVariables();
+
+            $ref = new \ReflectionFunction($closureVars["callback"]);
```

Comment:
looks like you have to have a very precise setup of callbacks used variables: `$callback`, `$request` and `$requestStack`. In addition to that there must be a specific instances and a provided callback must return some sort of iterable type.

Ideally, symfony's StreamedResponse callback should be compatible with response [example at RR docs ](https://roadrunner.dev/docs/http-resp-streaming/current/en). But since symfony streamed response defines callback as userland function without any return - such implementation with hardcoded variable values adds coupling with RR as web runner.

There are [multiple ways](https://dev.to/rubenrubiob/serve-a-file-stream-in-symfony-3ei3) on how to create a Symfony's StreamedResponse, but I cannot think of possible implementation on how to transform them to generators in an elegant way without coupling and hardcoding precise structure of variables..

Reply:
how to write functional tests then? Symfony wont print yielded values from function.

Leaving echo together with yield is also not an option since it will store the data to memory (`ob_start()`, `ob_get_clean()`)

Reply:
I mean that your provided example with `yield fread($stream, 1024);` won't work in a non RR environment.

Meaning that you are coupling you code implementation to RoadRunner and loosing flexibility which [runtime](https://symfony.com/doc/current/components/runtime.html) provides

Reply:
Ah, yes. Still, your argument loses meaning:

I said before, you would either need to use custom StreamedRespose that this bundle would provide or use Symfony's one and change echo's to yields.

**Both make you runtime dependant.**

By using runtime of choice, you are immediately being "held hostage" by it, no matter which one it is. Doesn't matter if it's the default one, RR, Swoole or other variants, you will always need to adjust your code in some way, if it's running in worker mode.

Reply:
Also, maybe Symfony would accept a PR that allows to pass callback that returns generators to `StreamedResponse`, if you have time for another PR :)

Reply:
Sure. I will look at this tomorrow. Do you also want one for the rest of them, or just this one?